src/devices/Tcs3472x/Tcs3472x.cs

```diff
@@ -0,0 +1,227 @@
+        }
+
+        /// <summary>
+        /// Set the integration (sampling) time for the sensor
+        /// </summary>
+        /// <param name="timeSeconds">Time in seconds for each sample. 0.0024 second(2.4ms) increments.Clipped to the range of 0.0024 to 0.6144 seconds.</param>
+        public void SetIntegrationTime(double timeSeconds)
+        {
+            if (timeSeconds <= 700)
+            {
+                if (_isLongTime)
+                {
+                    SetConfigLongTime(false);
+                }
+                _isLongTime = false;
+                var timeByte = (int)(0x100 - (timeSeconds / 0.0024));
+                timeByte = Math.Clamp(timeByte, 0, 255);
+                _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.ATIME), (byte)timeByte });
+                _integrationTimeByte = (byte)timeByte;
+            }
+            else
+            {
+                if (!_isLongTime)
+                {
+                    SetConfigLongTime(true);
+                }
+                _isLongTime = true;
+                var timeByte = (int)(0x100 - (timeSeconds / 0.029));
+                timeByte = Math.Clamp(timeByte, 0, 255);
+                _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.WTIME), (byte)timeByte });
+                _integrationTimeByte = (byte)timeByte;
+            }
+        }
+
+        private void SetConfigLongTime(bool setLong)
+        {
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.CONFIG), setLong ? (byte)(Registers.CONFIG_WLONG) : (byte)0 });
+        }
+
+        private void PowerOn()
+        {
+
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.ENABLE), (byte)Registers.ENABLE_PON });
+            Thread.Sleep(10);
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.ENABLE), (byte)(Registers.ENABLE_PON | Registers.ENABLE_AEN) });
+        }
+
+        private void PowerOff()
+        {
+            var powerState = I2cRead8(Registers.ENABLE);
+            powerState = (byte)(powerState & ~(byte)(Registers.ENABLE_PON | Registers.ENABLE_AEN));
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.ENABLE), powerState });
+        }
+
+        /// <summary>
+        /// Set/Clear the colors and clear interrupts
+        /// </summary>
+        /// <param name="state">true to set all interrupts, false to clear</param>
+        public void SetInterrupt(bool state)
+        {
+            SetInterrupt(InterruptState.All, state);
+        }
+
+        /// <summary>
+        /// Set/clear a specific interrupt persistence
+        /// </summary>
+        /// <param name="interupt">The percistence cycles</param>
+        /// <param name="state">True to set the interrupt, false to clear</param>
+        public void SetInterrupt(InterruptState interupt, bool state)
```

Comment:
how should the user use this? do you set interrup flag first and then check if interrup is clear with the other property? perhaps might be worth to add some comment about it - also might be worth renaming to something like:
StartWaitingFor...
Is...Ready

Reply:
It's about the number of cycle you wait to generate an interruption. So it's more to use instead of the previous one